python/pyarrow/io.pxi

```diff
@@ -39,13 +39,14 @@ cdef extern from "Python.h":
 
 
 cdef class NativeFile:
-
     def __cinit__(self):
-        self.is_open = False
+        self.closed = True
         self.own_file = False
```

Comment:
Should `own_file` be true for most of these? I stuck with the existing behavior, but this seems odd to me. For example `MemoryMappedFile` opens the file itself, why isn't it responsible for closing it on `__dealloc__`? Are we relying on a C++ destructor to handle that?

Reply:
I think this arose due to some wacky GC issues with HDFS file handles, I'd have to look back at the original commit that added this to remember. My prior would be that `__dealloc__` / `__del__` should close the file if it is open. With `MemoryMappedFile`, this is a no-op because the file descriptor is owned by the mapped buffer, whose lifetime may exceed the file object

Reply:
Ok. I'll leave that for a follow-up PR if that's ok. I believe this attribute is unnecessary for any of the existing file objects.